core/theme/Theme.php

```diff
@@ -0,0 +1,110 @@
+<?php
+
+namespace luya\theme;
+
+use luya\helpers\Json;
+use luya\helpers\StringHelper;
+use Yii;
+use yii\base\BaseObject;
+use yii\base\InvalidConfigException;
+
+/**
+ * Extend the yii2 theme component and build up the `$pathMap` depends on given \luya\theme\ThemeConfig which include parents inheritance.
+ *
+ * @see \luya\theme\ThemeConfig
+ * @author Bennet Klarhoelter <boehsermoe@me.com>
+ *
+ * @property-read string $layout
```

Comment:
We have not yet defined read or write properties, we could do so if we like. but i am note sure if every IDE follows those paths. until now we where using only `@property`. Let's discuss? Or maybe i make a small research whether property-read is widely accepted.

Reply:
Founded this https://github.com/yiisoft/yii2/pull/10989
So better not use it...